cmd/podmanV2/root.go

```diff
@@ -83,17 +134,62 @@ func syslogHook() {
 	}
 }
 
-func Execute() {
-	o := registry.NewOptions(rootCmd.Context(), &registry.EngineOptions)
-	if err := rootCmd.ExecuteContext(o); err != nil {
-		fmt.Fprintln(os.Stderr, "Error:", err.Error())
-	} else if registry.GetExitCode() == registry.ExecErrorCodeGeneric {
-		// The exitCode modified from registry.ExecErrorCodeGeneric,
-		// indicates an application
-		// running inside of a container failed, as opposed to the
-		// podman command failed.  Must exit with that exit code
-		// otherwise command exited correctly.
-		registry.SetExitCode(0)
+func rootlessHook() {
+	if rootless.IsRootless() {
+		logrus.Error("rootless mode is currently not supported. Support will return ASAP.")
 	}
-	os.Exit(registry.GetExitCode())
+	// ce, err := registry.NewContainerEngine(rootCmd, []string{})
+	// if err != nil {
+	// 	logrus.WithError(err).Fatal("failed to obtain container engine")
+	// }
+	// ce.SetupRootLess(rootCmd)
+}
+
+func rootFlags(opts entities.PodmanConfig, flags *pflag.FlagSet) {
+	// V2 flags
+	flags.StringVarP(&opts.Uri, "remote", "r", "", "URL to access Podman service")
+	flags.StringSliceVar(&opts.Identities, "identity", []string{}, "path to SSH identity file")
+
+	// Override default --help information of `--version` global flag}
+	var dummyVersion bool
+	flags.BoolVar(&dummyVersion, "version", false, "Version of Podman")
+
+	cfg := opts.Config
+	flags.StringVar(&cfg.Engine.CgroupManager, "cgroup-manager", cfg.Engine.CgroupManager, opts.CGroupUsage)
+	flags.StringVar(&opts.CpuProfile, "cpu-profile", "", "Path for the cpu profiling results")
+	if err := flags.MarkHidden("cpu-profile"); err != nil {
+		logrus.WithError(err).Warn("unable to mark cpu-profile flag as hidden")
+	}
+	flags.StringSliceVar(&cfg.Engine.ConmonPath, "conmon", []string{}, "Path of the conmon binary")
```

Comment:
This should not be a string slice. We accept one conmon path, and only one.

Reply:
fixed